### ISSUE TITLE
Invalid Vehicle Type when Building Track Design with Unresearched Vehicle Type + Phantom Guest Payment Crash

### ISSUE THREAD

**Issue by Androgeos** [OpenRCT2/OpenRCT2#3467]
**OS: Windows 10 (build 10586)**
**Version: 0.0.5-develop**
**Commit: 7656cfd & b5706a5 (both from https://openrct2.org/downloads)**

OpenRCT2 appears to crash if it attempts to operate a track design using a vehicle from a ride design that has not been researched yet or is not available yet on a ride design of a similar track type (eg. building and operating a Chesire Cat track design via Car Ride when Chesire Cats is not researched).
Attempting to reproduce the crash may cause the game to freeze instead.
- [ ] Reproducible in RCT2 (vanilla)
- [ ] Multiplayer

**Steps to reproduce:**
1. Enable "Select rides by track type (like in RCT1)" and set Current Theme to RCT1 under Options -> Controls and interface (ie. gears tab)
2. Enable "Allow building in pause mode" under Cheats -> Ride cheats (ie. Spiral Slide tab)
3. Set Current Theme to RCT1 under Options -> Controls and interface (ie. gears tab)
4. Load "Bumbly Beach". Scenario file is attached here:
[Bumbly Beach.zip](https://github.com/OpenRCT2/OpenRCT2/files/244031/Bumbly.Beach.zip)
5. Pause the game.
6. Close Big Dipper and effect the following changes to the track design:
Replace the last 3 sections of station platform with 2 brake sections set to 28 km/h and a block brake section, with the latter being built right before the remaining station platform
Replace the 5 sections of the dip before the last drop with a water spash and replace the medium slope track before the water splash with a slope to level piece so that the track connects
Replace the left slope at the far end of the ride (ie. when the tracks turn away from the beach) with a curved left slope.
7. Test Big Dipper and make the necessary alterations to the footpaths
8. Relocate Big Wheel to the 1x4 space after the first drop of Big Dipper, leaving 1 square to place the Ride Exit between the short edge of Big Wheel and footpath. Before setting down Big Wheel at the new location, rotate it twice so that it rotates in the other direction. Place Big Wheel down and place the entrance and queue line. Test it.
9. Build a Haunted House beside Big Dipper at the section where the track turns back towards the station. Test it.
10. Build a Twist beside Information Kiosk 1. Test it.
11. Build a Drinks Stall
12. Build a Fries Stall
13. Build a Toilets and set admission price to 10 cents.
14. Build "Chessie Cat Cars" (you will need to borrow at least $1000 more from the bank to do this). Track design attached:
[Chessie Cat Cars.zip](https://github.com/OpenRCT2/OpenRCT2/files/244028/Chessie.Cat.Cars.zip)
14a. Your park should look like the first screenshot under the Screenshots / Video section below at this point.
15. Test "Chessie Cat Cars".
16. Unpause the game.

The crash occurs approximately 1 second after executing Step 16, or otherwise freezes the game. It also generates a wall of error messages similar to the 6 lines shown in the second screenshot under the Screenshots / VIdeo section.
It is also possible that the game may continue to run as though there is no issue if construction of Chessie Cat Cars is done first, and while the game is unpaused; however, the Command Prompt window will still record 4 error lines identical to those in the second screenshot in the "Screenshots / Video" section below.

**Dump file**
[8a21fda2-2c40-4e0a-833c-5da9d7a88efb.dmp.zip](https://github.com/OpenRCT2/OpenRCT2/files/244021/8a21fda2-2c40-4e0a-833c-5da9d7a88efb.dmp.zip)

**Screenshots / Video:**
![scr3](https://cloud.githubusercontent.com/assets/18898491/14940980/bd8709b0-0fbe-11e6-811b-82c0f94abbce.png)
Screenshot of a reproduction of the issue just before the game froze. Note the Cat Cars on the right.

![cmd1](https://cloud.githubusercontent.com/assets/18898491/14940979/bd85aa66-0fbe-11e6-921e-171a006f6e63.png)
Background Command-Line window as Chessie Cat Cars was built or tested.

**Save game:**
[8a21fda2-2c40-4e0a-833c-5da9d7a88efb.sv6.zip](https://github.com/OpenRCT2/OpenRCT2/files/244022/8a21fda2-2c40-4e0a-833c-5da9d7a88efb.sv6.zip)

**Comment by jensj12** [comment]
Does it still crash if you skip step 14 and/or 15? If not, the issue is indeed the cat ride and just posting steps 1-5 and 14-16 would be enough to reproduce the issue for the devs. No need to post the other 8 (time-consuming) steps in between.

**Comment by Androgeos** [comment]
...well, this is interesting. It appears that Chessie Cat Cars (and, by extension, this issue) _isn't_ causing the crash, because it still crashed when I _didn't_ build Chessie Cat Cars.

On the other hand, if I start the scenario and _don't_ do anything, it runs just fine.

The crash seems to occur when a guest pays for something after I've performed all steps regardless of whether I perform Steps 14 and 15 or not. However, I'm not able to determine which guest is paying for what; my own observation of the following save file I made (which coincidentally was done moments before the crash) doesn't visually show any guest paying for anything at the point of the crash.
[Bumbly Beach.sv6.zip](https://github.com/OpenRCT2/OpenRCT2/files/244112/Bumbly.Beach.sv6.zip)

Another thing to note: I cannot replicate the crash on vanilla RCT2 at all, and if I disable the cheat to build while in pause mode, the game doesn't crash even with Chessie Cat Cars built, although the background Command Prompt window still shows the same error message:
![cmd2](https://cloud.githubusercontent.com/assets/18898491/14942033/965aa372-0fe3-11e6-9ddf-85e43b0eb7b0.png)

Here's my DMP and SV6 of the above Bumbly Beach.sv6 at the point of crash:
[ec1e713f-ddea-4513-9101-bd9370ee7a19.dmp.zip](https://github.com/OpenRCT2/OpenRCT2/files/244113/ec1e713f-ddea-4513-9101-bd9370ee7a19.dmp.zip)
[ec1e713f-ddea-4513-9101-bd9370ee7a19.sv6.zip](https://github.com/OpenRCT2/OpenRCT2/files/244114/ec1e713f-ddea-4513-9101-bd9370ee7a19.sv6.zip)

**Comment by janisozaur** [comment]
Oh wow, 16 steps to reproduce it. I won't even try.

@Nubbie can you check what in the dumps here please?

**Comment by Nubbie** [comment]
Did not try the 16 steps either, it's good that it's extremely accurate but presume it's just the last part which is important, don't want to be rude but sounds a bit like a tutorial for a lego set :p 

--------------------

@janisozaur 
Commit used; 7656cfd 
Dumpfile; ec1e713f-ddea-4513-9101-bd9370ee7a19.dmp

Error message;
Unhandled exception thrown: read access violation.
other_peep was 0x20E62BC.

**Call stack**
```
>	openrct2.dll!remove_peep_from_queue(rct_peep * peep=0x010ecebc) Line 1584	C
 	openrct2.dll!peep_interact_with_path(rct_peep * peep=0x010ecebc, short x, short y, rct_map_element * map_element=0x00f98e20) Line 7298	C
 	openrct2.dll!sub_693C9E(rct_peep * peep=0x010ecebc) Line 8405	C
 	openrct2.dll!peep_update_queuing(rct_peep * peep=0x010ecebc) Line 4046	C
 	[External Code]	
```

**Locals;**
```
+		other_peep	0x020e62bc {sprite_identifier=??? var_01=??? next_in_quadrant=??? ...}	rct_peep *
+		peep	openrct2.exe!0x010ecebc (load symbols for additional information) {sprite_identifier=1 '\x1' var_01=...}	rct_peep *
		ride	Variable is optimized away and not available.	
```

**Autos;**
```
+		peep	openrct2.exe!0x010ecebc (load symbols for additional information) {sprite_identifier=1 '\x1' var_01=...}	rct_peep *
		peep->x	2608	short
		peep->y	2753	short
```

**Comment by janisozaur** [comment]
I assume this is fixed by 6db3828f
